src/storage/src/sink/kafka.rs

```diff
@@ -1142,13 +1143,59 @@ fn encode_collection<G: Scope>(
                 .map(|(desc, _indices)| desc.clone());
             let value_desc = connection.value_desc;
 
-            let encoder: Box<dyn Encode> = match connection.format {
-                KafkaSinkFormat::Avro {
-                    key_schema,
-                    value_schema,
+            let key_encoder: Option<Box<dyn Encode>> = if let Some(desc) = key_desc {
+                match connection.format.key_format {
```

Comment:
Is it ever possible to have a key_format specified without a key description? Why do we need the `if let Some(desc).. ` part?

Reply:
> Is it ever possible to have a key_format specified without a key description?

No this should be enforced during planning such that they are tied together

> Why do we need the if let Some(desc)..  part?

Just to unwrap the `desc` out of the option. I could probably refactor this to use a combined `match (key_desc, connection.format.key_format)` statement if that's easier to read